File: capgemini/batch-3/08-02-24/MissingNumber.py


def getMissingNumber(n,arr):
    ans = 0
    for i in range(1,n+1):
        ans = ans^i
    for e in arr:
        ans = ans^e
    return ans

def main():
    n = 10
    arr =[2,1,4,3,7,10,9,6,5]
    ans = getMissingNumber(n,arr)
    print(ans)

main()

File: CMR-EC/P3/31-08-2024/MissingNumber.java
class MissingNumber{
    public static void main(String[] args) {
        int n = 10;
        int[] arr = {1,7,10,8,9,3,2,5,6};
        int ans = getMissingNumber(arr,n);
        System.out.println(ans);
    }
    private static  int getMissingNumber(int[] arr, int n){
        int missingNumber = 0;
        for(int i=1;i<=n;i++)
            missingNumber = missingNumber ^ i;
        for(int e : arr)
            missingNumber = missingNumber ^ e;

        return missingNumber;
    }
}